src/AppBundle/Controller/TodoController.php

```diff
@@ -0,0 +1,242 @@
+    }
+
+    /**
+     * @Route("/todo", name="delete_todo")
+     * @Method({"DELETE"})
+     *
+     * Deletes all of the existing To do entries for this user.
+     *
+     * @return JsonResponse
+     */
+    public function deleteTodoAction()
+    {
+        $itemsDeleted = 0;
+        try {
+            $collection = $this->dm()->getDocumentCollection('AppBundle:Todo');
+            //The result from the mongodb operation
+            $mongoResult = $collection->remove(array('userId' => $this->getUser()->getIdString()));
+            if (is_array($mongoResult) && isset($mongoResult['n']))
+                $itemsDeleted = intval($mongoResult['n']);
+            $this->dm()->flush();
+        } catch (Exception $e) {
+            return new JsonResponse('' . $e, 500);
+        }
+
+        return new JsonResponse('Successfully removed all (' . $itemsDeleted . ') todo items for this user.');
+    }
+
+    /**
+     * @Route("/todo", name="post_todo")
+     * @Method({"POST"})
+     *
+     * Creates a new to do, and responds with the ID value (string)
+     *
+     * @param Request $request
+     * @return JsonResponse
+     *
+     * request
+     *  body:
+     *      type: object
+     *      properties:
+     *          title:
+     *              type: string
+     *              required: true
+     *              description: The new to do item. Must not be empty.
+     * responses:
+     *  200:
+     *      OK
+     *  400:
+     *      Bad Request
+     *
+     *
+     */
+    public function postTodoAction(Request $request)
+    {
+        $body = null;
+        try {
+            $body = $this->bodyAsJSON($request);
+            //If title is unset or null, or isnt a string
+            if (!isset($body['title']) || !is_string($body['title']))
+                throw new JSONControllerException("Required parameter 'title' was unset or not a string value.");
+        } catch (JSONControllerException $e) {
+            return new JsonResponse('' . $e, 400);
+        }
+        //$body isnt null and title is set
+
+        $todo = new Todo();
+        $todo->setCompleted(false);
+        $todo->setTitle($body['title']);
+        $todo->setUserId($this->getUser()->getIdString());
+
+        $this->dm()->persist($todo);
+        $this->dm()->flush();
+
+        return new JsonResponse('' . $todo->getId());
+    }
+
+    /**
+     * @Route("/todo/{id}", name="get_todo_id")
+     * @Method({"GET"})
+     *
+     * Gets a specific item.
+     *
+     * @param string $id
+     * @return JsonResponse
+     */
+    public function getTodoIdAction($id)
+    {
+        $item = $this->get('doctrine_mongodb')
+            ->getRepository('AppBundle:Todo')
+            ->find($id);
+
+        if (!$item) {
+            return new JsonResponse('Not Found', 404);
+        }
+
+        //if we are forbidden (we are authorized)
+        if ($item->getUserId() != $this->getUser()->getIdString()) {
+            return new JsonResponse('Forbidden', 403);
```

Comment:
It's good practice to return a 404 here as well. Otherwise an unauthorised attacker has just gained information as to whether a todo is in your system. It's the same reason you shouldn't say whether or not a username is in your system when someone fails to login.